### PR TITLE
Update networking files in order to be merged into master branch

### PR DESCRIPTION
**Pull request by a7ehuo**
Signed-off-by: Annabelle Huo <Annabelle.Huo@ibm.com>

### CODE REVIEW

File: runtime/compiler/net/ProtobufTypeConvert.hpp [hunk 1]
```diff
@@ -108,13 +106,13 @@ namespace JITServer
          {
          char data[sizeof(T)];
          const std::string &strVal = msg->bytes_v();
-         memcpy(data, &strVal[0], sizeof(T));
+         memcpy(data, strVal.c_str(), sizeof(T));
```
Comment (mpirvu) [inline runtime/compiler/net/ProtobufTypeConvert.hpp]: I would argue that `strVal.data()` is a better choice here

File: runtime/compiler/build/rules/common.mk [hunk 2]
```diff
@@ -25,7 +25,7 @@ cleanobjs: jit_cleanobjs
 cleandeps: jit_cleandeps
 cleandll: jit_cleandll
 ifneq ($(JITSERVER_SUPPORT),)
-    proto: proto_clean protoc
+proto: protoc
```
[the remark this answers is not in the document]
Comment (a7ehuo) [reply runtime/compiler/build/rules/common.mk]: proto_clean is actually not defined. proto build artifacts are cleaned by `RULE.proto` by appending them to `jit_cleanobjs`.

File: runtime/compiler/build/rules/gnu/common.mk [hunk 3]
```diff
@@ -70,8 +70,8 @@ jit_cleanobjs::
 	rm -f $(JIT_PRODUCT_BUILDNAME_SRC)
 
 ifneq ($(JITSERVER_SUPPORT),)
-   protoc: $(PROTO_GEN_DIR)/compile.pb.h
-   $(call RULE.proto,$(PROTO_GEN_DIR)/compile,$(PROTO_DIR)/compile.proto)
+    protoc: $(PROTO_GEN_DIR)/compile.pb.h
+        $(call RULE.proto,$(PROTO_GEN_DIR)/compile,$(PROTO_DIR)/compile.proto)
```
[the remark this answers is not in the document]
Comment (a7ehuo) [reply runtime/compiler/build/rules/gnu/common.mk]: Commands need to be indented wth TABs. Here is makefile syntax and shouldn't be indented with TABs. Tried intended with TABs and it failed to compile.

**Comment by mpirvu** [comment]
Jenkins test sanity.functional+jitaas xlinux jdk8 depends eclipse/openj9-omr#jitaas

**Comment by mpirvu** [comment]
@a7ehuo Actually, I have one question: there are a few files that have no changes, but I think that the protection flags on them changed. Was that intentional?

**Comment by mpirvu** [comment]
The only failure is the known "cmdLineTester_gcsuballoctests_0" so this change set is ready to be merged.

**Comment by a7ehuo** [comment]
To be consistent with other files, the permission on those .cpp and .h files is updated from `-rwxr-xr-x ` to `-rw-r--r--`.

**Pull request merged**